tornado/netutil.py

```diff
@@ -160,7 +160,14 @@ def bind_sockets(port, address=None, family=socket.AF_UNSPEC,
             sockaddr = tuple([host, bound_port] + list(sockaddr[2:]))
 
         sock.setblocking(0)
-        sock.bind(sockaddr)
+        try:
+            sock.bind(sockaddr)
+        except socket.error as e:
+            # Some setups (e.g. Travis-CI with "localhost") might return
```

Comment:
This seems to be a bug in travis (or something in container-land?) - if you can't bind to `::1`, then `/etc/hosts` really shouldn't specify that `localhost` has that address. 

I don't like swallowing this error in production code paths; I'd be more comfortable if it were confined to test-only code. We work around this differently in tornado.testing, by binding to `127.0.0.1` instead of resolving `localhost` (this was originally motivated by silly firewall behavior on macOS, but it helps here too). 

I think I'd prefer to see this problem addressed by fixing the broken `/etc/hosts` file or explicitly binding to an address other than `localhost` (although we have other similar workarounds above for the many ways in which ipv6 can be incompletely available, so maybe this hack isn't so bad). 

There's also `AI_ADDRCONFIG`, which looks like it might help here, but we had it before and removed it in #593.

Reply:
I understand your sentiment. OTOH, the problem is that we would need to add some special-casing for Travis in various places (and we may not be the only project having to do this), without any clear advantage in doing so.

As for the broken `/etc/hosts` file, I wonder if other systems have such a setup. Working such misconfigurations in Tornado doesn't sound unreasonable, especially as users often don't have a hand on the system configuration.

Reply:
> I understand your sentiment. OTOH, the problem is that we would need to add some special-casing for Travis in various places

Well, I'd hope that travis would fix this on their side. 

> As for the broken /etc/hosts file, I wonder if other systems have such a setup.

I've seen this be broken in such a way that `localhost` resolves to both ipv4 and ipv6, but binding to ipv6 returns `EAFNOSUPPORT`. I haven't seen `EADDRNOTAVAIL` elsewhere.

My concern with this workaround is that it is too broad: This is also the error you'd get if you were trying to bind a specific IP address but you mistyped it. That's not an error that should pass silently.